demo/image_creation.sh

```diff
@@ -1,125 +1,162 @@
+echo "*********************************************************************"
+TARGET_DISK=$(az disk show --ids "$(az vm show -g "$RG" -n "$VM_NAME" | jq .storageProfile.osDisk.managedDisk.id -r)" | jq .name -r)
+az snapshot create -g "$RG" -n "$VM_NAME"-snapshot --source "$TARGET_DISK"
+
+IMAGE_VERSION=$(date '+%Y.%m%d.%H%M%S')
+GALLERY_NAME=testgalleryazinit
+GALLERY_DEFINITION=testgalleryazinitdef
+GALLERY_RG=testgalleryazinitrg
+SNAPSHOT_ID=$(az snapshot show -n "$VM_NAME"-snapshot -g "$RG" --query id --output tsv)
+
+# If BASE_IMAGE == canonical:0001-com-ubuntu-server-jammy:22_04-lts-gen2:latest,
+# GALLERY_PUBLISHER = canonical
+# GALLERY_OFFER = 0001-com-ubuntu-server-jammy
+# GALLERY_SKU = 22_04-lts-gen2
+GALLERY_PUBLISHER="$(echo "$BASE_IMAGE" | cut -f1 -d:)"
+GALLERY_OFFER="$(echo "$BASE_IMAGE" | cut -f2 -d:)"
+GALLERY_SKU="$(echo "$BASE_IMAGE" | cut -f3 -d:)"
+
+az group create --location "$LOCATION" --resource-group "$GALLERY_RG"
+
+az sig create --resource-group "$GALLERY_RG" --gallery-name "$GALLERY_NAME"
+
 echo "*********************************************************************"
-echo "Capturing OsDisk snapshot of vm $vm with image $image"
+echo "Creating image definition of $BASE_IMAGE"
 echo "*********************************************************************"
-target_disk=$(az disk show --ids $(az vm show -g $rg -n $vm | jq .storageProfile.osDisk.managedDisk.id -r) | jq .name -r)
-az snapshot create -g $rg -n $vm-snapshot --source $target_disk
+az sig image-definition create --resource-group "$GALLERY_RG" --gallery-name "$GALLERY_NAME" --gallery-image-definition "$GALLERY_DEFINITION" --offer "$GALLERY_OFFER" --publisher "$GALLERY_PUBLISHER" --sku "$GALLERY_SKU" --os-type linux --os-state Specialized --features SecurityType="$SECURITY_TYPE"
 
-version=$(date '+%Y.%m%d.%H%M%S')
-gallery=testgalleryagent
-definition=testgallery-gen1
-gallery_rg=temp-rg-rust-agent-testing
-snapshot_id=$(az snapshot show -n $vm-snapshot -g $rg --query id --output tsv)
 echo "*********************************************************************"
-echo "Publishing image version $version to $gallery/$definition"
+echo "Publishing image version $IMAGE_VERSION to $GALLERY_NAME/$GALLERY_DEFINITION"
 echo "*********************************************************************"
-az sig image-version create -g $gallery_rg --gallery-name $gallery --gallery-image-definition $definition --gallery-image-version $version --os-snapshot $snapshot_id --target-regions "eastus" --replica-count 1
-if [[ $? -eq 0 ]]
+if az sig image-version create -g $GALLERY_RG --gallery-name $GALLERY_NAME --gallery-image-definition $GALLERY_DEFINITION --gallery-image-version "$IMAGE_VERSION" --os-snapshot "$SNAPSHOT_ID" --target-regions "$LOCATION" --replica-count 1;
 then
     echo "Image publishing finished"
     echo "Deleting staging resource group"
-    az group delete -g $rg --yes --no-wait
+    az group delete -g "$RG" --yes --no-wait
     echo "Done"
 fi
```

Comment:
if the image version creation fails, we should print out some error, and remind the user that the resource group is kept for debugging purposes

Reply:
Done